cuda_core/tests/conftest.py

```diff
@@ -5,12 +5,28 @@
 # this software. Any use, reproduction, disclosure, or distribution of
 # this software and related documentation outside the terms of the EULA
 # is strictly prohibited.
+try:
+    from cuda.bindings import driver
+except ImportError:
+    from cuda import cuda as driver
 
-from cuda.core.experimental._device import Device
+from cuda.core.experimental import Device
+from cuda.core.experimental import _device
+from cuda.core.experimental._utils import handle_return
 import pytest
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="function")
 def init_cuda():
     device = Device()
     device.set_current()
-    
\ No newline at end of file
+    yield
+    handle_return(driver.cuCtxPopCurrent())
+    with _device._tls_lock:
+        del _device._tls.devices
+
+@pytest.fixture(scope="function")
+def deinit_cuda():
+    yield
+    handle_return(driver.cuCtxPopCurrent())
+    with _device._tls_lock:
+        del _device._tls.devices
```

Comment:
nit: let's try not to duplicate code
```suggestion
    _device_unset_current()

def _device_unset_current():
    handle_return(driver.cuCtxPopCurrent())
    with _device._tls_lock:
        del _device._tls.devices

@pytest.fixture(scope="function")
def deinit_cuda():
    yield
    _device_unset_current()
```